moshi/src/test/java/com/squareup/moshi/JsonCodecFactory.java

```diff
@@ -29,8 +29,10 @@
       Buffer buffer;
 
       @Override public JsonReader newReader(String json) {
-        Buffer buffer = new Buffer().writeUtf8(json);
-        return JsonReader.of(buffer);
+        try (Buffer buffer = new Buffer()) {
```

Comment:
This is incorrect as it will call `close()` on the `Buffer` before returning a `JsonReader` that wraps it and intends to then read from it.

Reply:
It doesn't break anything because `Buffer` ignores `close()`, but it's still semantically wrong.